lib/artifact.js

```diff
@@ -136,29 +148,32 @@ module.exports = (bluebird, fs, logger, request, requestOptions, requestPromise,
             options.url = this.uri;
             logger.debug(`GET request options: ${JSON.stringify(options, null, 4)}`);
 
-            return new Promise((resolve, reject) => {
+            return new Promise((resolve) => {
```

Comment:
I don't think you should be calling a `new Promise` using Node's native Promises. You'd want to be using bluebird, since you're already injecting it, it's faster, and you limit your dependencies by only using one promise library.